lib/kitchen/driver/ec2.rb

```diff
@@ -424,7 +424,31 @@ def expand_config(conf, key)
       def submit_spots
         configs = [config]
         expanded = []
-        keys = %i{instance_type subnet_id}
+        keys = %i{instance_type}
+
+        unless config[:subnet_filter]
+          # => Use explicitly specified subnets
+          keys << :subnet_id
+        else
+          # => Enable cascading through matching subnets
```

Comment:
it seems like this whole part could be extracted into a method and mocked / tested to some degree. I know we have massive methods already in this driver, but I'd like to try to prevent the sprawl going forward and slowly tackle the sprawl we've already shipped at a later date.